From e49a95040f0909608f62e3dd9e435bf58292553b Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Fri, 24 Aug 2018 17:26:11 +0200
Subject: [PATCH] Fixed build under GCC 8.2.1. Added missing includes.

Signed-off-by: Vitaly Zaitsev <vitaly@easycoding.org>
---
 VoIPController.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/VoIPController.cpp b/VoIPController.cpp
index 4032411..66416b1 100644
--- a/VoIPController.cpp
+++ b/VoIPController.cpp
@@ -26,6 +26,7 @@
 #include <stdexcept>
 #include <algorithm>
 #include <inttypes.h>
+#include <cfloat>
 
 
 
-- 
2.17.1

