From e0e347b3c1a26754b250d02a78488cf9093a6d7b Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 9 Jan 2020 13:18:09 +0100
Subject: [PATCH] Fixed build of libtgvoip under Fedora as shared library.

---
 Makefile.am  |  7 +++----
 configure.ac | 13 ++++++++++++-
 2 files changed, 15 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 03c8866..0320457 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -26,8 +26,7 @@ audio/Resampler.cpp \
 os/posix/NetworkSocketPosix.cpp \
 video/VideoSource.cpp \
 video/VideoRenderer.cpp \
-video/ScreamCongestionController.cpp \
-json11.cpp
+video/ScreamCongestionController.cpp
 
 TGVOIP_HDRS = \
 TgVoip.h \
@@ -55,7 +54,6 @@ os/posix/NetworkSocketPosix.h \
 video/VideoSource.h \
 video/VideoRenderer.h \
 video/ScreamCongestionController.h \
-json11.hpp \
 utils.h
 
 if TARGET_OS_OSX
@@ -751,6 +749,7 @@ else
 CFLAGS += -DTGVOIP_NO_DSP
 endif
 
+libtgvoip_la_LDFLAGS = -version-number @LIBTGVOIP_MAJOR_VERSION@:@LIBTGVOIP_MINOR_VERSION@:@LIBTGVOIP_PATCH_VERSION@
 libtgvoip_la_SOURCES = $(SRC) $(TGVOIP_HDRS)
 tgvoipincludedir = $(includedir)/tgvoip
 nobase_tgvoipinclude_HEADERS = $(TGVOIP_HDRS)
@@ -759,4 +758,4 @@ CXXFLAGS += -std=gnu++0x $(CFLAGS)
 if TARGET_OS_OSX
 OBJCFLAGS = $(CFLAGS)
 OBJCXXFLAGS += -std=gnu++0x $(CFLAGS)
-endif
\ No newline at end of file
+endif
diff --git a/configure.ac b/configure.ac
index 222f541..cfad4ef 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,8 +1,14 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
+# Define version information
+m4_define([libtgvoip_major_version], 2)
+m4_define([libtgvoip_minor_version], 4)
+m4_define([libtgvoip_patch_version], 4)
+m4_define([libtgvoip_version],[libtgvoip_major_version.libtgvoip_minor_version.libtgvoip_patch_version])
+
 AC_PREREQ([2.69])
-AC_INIT([libtgvoip], [2.4.4], [https://github.com/grishka/libtgvoip/issues])
+AC_INIT([libtgvoip], [libtgvoip_version], [https://github.com/grishka/libtgvoip/issues])
 AC_CONFIG_SRCDIR([config.h.in])
 AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([subdir-objects])
@@ -21,9 +27,14 @@ AC_CHECK_LIB([crypto], [SHA1], [], [AC_MSG_FAILURE([libssl-dev is required but n
 AC_CHECK_LIB([m], [floorf])
 AC_CHECK_LIB([opus], [opus_decoder_create], [], [AC_MSG_FAILURE([libopus-dev is required but not found])])
 AC_CHECK_LIB([pthread], [pthread_create])
+AC_CHECK_LIB([json11], [main], [], [AC_MSG_FAILURE([json11-dev is required but not found])])
 
 AC_CANONICAL_HOST
 
+AC_SUBST(LIBTGVOIP_MAJOR_VERSION, libtgvoip_major_version)
+AC_SUBST(LIBTGVOIP_MINOR_VERSION, libtgvoip_minor_version)
+AC_SUBST(LIBTGVOIP_PATCH_VERSION, libtgvoip_patch_version)
+
 AS_CASE([$host_cpu],
 	[i?86], [cpu_x86=yes],
 	[x86_64], [cpu_x86=yes],
-- 
2.24.1

