From 100f7ecbd889a94742c378b726dc46bfbc27ad96 Mon Sep 17 00:00:00 2001
From: Grishka <grishka93@gmail.com>
Date: Thu, 19 Jul 2018 14:52:38 +0300
Subject: [PATCH] Fixed crash

---
 threading.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/threading.h b/threading.h
index dbba1f9..cef6d3c 100644
--- a/threading.h
+++ b/threading.h
@@ -75,6 +75,7 @@ namespace tgvoip{
 	public:
 		Thread(MethodPointerBase* entry, void* arg) : entry(entry), arg(arg){
 			name=NULL;
+			thread=NULL;
 		}
 
 		virtual ~Thread(){
@@ -86,7 +87,8 @@ namespace tgvoip{
 		}
 
 		void Join(){
-			pthread_join(thread, NULL);
+			if(thread)
+				pthread_join(thread, NULL);
 		}
 
 		void SetName(const char* name){
@@ -242,6 +244,7 @@ namespace tgvoip{
 	public:
 		Thread(MethodPointerBase* entry, void* arg) : entry(entry), arg(arg){
 			name=NULL;
+			thread=NULL;
 		}
 
 		~Thread(){
@@ -253,6 +256,8 @@ namespace tgvoip{
 		}
 
 		void Join(){
+			if(!thread)
+				return;
 #if !defined(WINAPI_FAMILY) || WINAPI_FAMILY!=WINAPI_FAMILY_PHONE_APP
 			WaitForSingleObject(thread, INFINITE);
 #else
