From 797768c6c278bee8a1fa361e3c9bc700a4764c3c Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Sat, 6 Jun 2020 14:16:01 +0200
Subject: [PATCH 2/2] Incremented C++ standard version.

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 3c23a74..1cec43b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -758,7 +758,7 @@ nobase_tgvoipinclude_HEADERS = $(TGVOIP_HDRS)
 
 pkgconfig_DATA = tgvoip.pc
 
-CXXFLAGS += -std=gnu++0x $(CFLAGS)
+CXXFLAGS += -std=gnu++17 $(CFLAGS)
 if TARGET_OS_OSX
 OBJCFLAGS = $(CFLAGS)
 OBJCXXFLAGS += -std=gnu++0x $(CFLAGS)
-- 
2.26.2

